### PR TITLE
Removed dependency on OrleansRuntime.dll in OrleansProvider.dll

### PR DESCRIPTION
**Pull request by gabikliot**
Removed dependency on OrleansRuntime.dll in OrleansProvider.dll
Moved the silo specific logic from PersistentStreamProvider, which is also used on the client, into SiloProviderRuntime, thus making PersistentStreamProvider be able to run on the client without dependency on OrleansRuntime.dll.

### CODE REVIEW

File: src/Orleans/Orleans.csproj [hunk 1]
```diff
@@ -353,15 +356,13 @@
...
+    <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <PropertyGroup>
     <CodeGenToolReferences>System.Data.dll;$(SolutionDir)packages\Microsoft.Data.Services.Client.5.6.0\lib\net40\Microsoft.Data.Services.Client.dll;</CodeGenToolReferences>
   </PropertyGroup>
-  <Target Name="GenerateVersionNumber" BeforeTargets="VersionInit" 
-          Condition="'$(BuildingInsideVisualStudio)' != 'true'" 
-          Outputs="$(SolutionDir)Build\Version.txt">
+  <Target Name="GenerateVersionNumber" BeforeTargets="VersionInit" Condition="'$(BuildingInsideVisualStudio)' != 'true'" Outputs="$(SolutionDir)Build\Version.txt">
```
Comment (gabikliot) [inline src/Orleans/Orleans.csproj]: I did not make any changes here. VS just reformatted the lines automatically.

File: src/OrleansProviders/OrleansProviders.csproj [hunk 2]
```diff
@@ -110,10 +110,6 @@
       <Project>{e782dd19-51f7-4f66-8217-bacac33767e4}</Project>
       <Name>ClientGenerator</Name>
     </ProjectReference>
-    <ProjectReference Include="..\OrleansRuntime\OrleansRuntime.csproj">
```
Comment (gabikliot) [inline src/OrleansProviders/OrleansProviders.csproj]: Removing this dependency was the goal of this PR.

**Comment by sergeybykov** [comment]
This was for https://github.com/dotnet/orleans/issues/185.

File: src/Orleans/Orleans.csproj (hunk 1 above)
Comment (jthelin) [inline src/Orleans/Orleans.csproj]: BTW, if you are using SourceTree app it has really nice feature that lets you selectively "revert" parts of a change in a file before you do the local commit. 
So for example you could remove this part of the change in this file but leave the other change below intact. 
See the lower right panel of the SourceTree commit form.

Comment (gabikliot) [inline src/Orleans/Orleans.csproj]: Thanks! Will use next time.

**Pull request merged**